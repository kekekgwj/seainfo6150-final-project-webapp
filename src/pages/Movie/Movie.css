.movie-wrapper {
    color: #111;
    width: 800px;
    margin: 0 auto;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.movie-content {
    min-height: 400px;
}

.movie-content h1 {

}
.movie-info {
    display: flex;
}
.movie-poster-wrapper {
    max-width: 200px;
    overflow: hidden;
}
.movie-poster {
    width: 200px;
}
.movie-detail {
    max-width: 400px;
    overflow: hidden;
    margin-left: 20px;
}
.label {
    font-size: 15px;
    line-height: 150%;
    color: #666666
}
.text {
    color: #111;
    font-size: 15px;
    margin-left: 5px;
}
.movie-rating-container {
    float: left;
    width: 155px;
    margin: 2px 0 0 100px;
    padding: 0 0 0 15px;
    border-left: 1px solid #eaeaea;
    color: #999;
}

.rating-text {
    display: inline-block;
    color: black;
    font-size: 32px;
    line-height: 32px;
}

.rating-wrapper {
    display: flex;
    align-items: center;
}

.movie-brief-title {
    color: orange;
}

.movie-brief-text {
    word-break: normal;
    word-wrap: break-word;
    font-size: 13px;
    color: #111;
}

.movie-review-title {
    color: orange
}

.movie-add-review {
    display: flex;
    align-items: center;
    font-size:  13px;
    background: orange;
    width: 180px;

}
.movie-add-review-text {
    position: relative;
    left: 10px;

}