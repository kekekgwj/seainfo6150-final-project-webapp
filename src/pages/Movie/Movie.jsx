import React, { useEffect, useState } from "react";
import "./Movie.css";
import star from "../../assets/star.png";
import pen from "../../assets/pen.png";
import ReviewList from "../../components/ReviewList/ReviewList";
import { useParams } from "react-router";
import { getMovieData } from "../../services";
import { useHistory } from 'react-router-dom';

const Movie = () => {
    const [data, setData] = useState({});
    const { id } = useParams();

    useEffect( () => {
        getMovieData(id)
        .then(res => {
            setData(res);
        })
        .catch(err => {
           setData({});
        });
    }, [id])
    const {info, reviews } = data;
    const history = useHistory();
    const addReview = () => {
        history.push({pathname:`/new_review/${id}`, state:{id, name: info.name}})
    }
    
    return (
        <div className="movie-wrapper">
            <div className="movie-content">
                {info && 
                    <div>
                    <h1>{info.name.toUpperCase()}</h1>
                    <div className="movie-intro">
                        <div className="movie-info">
                            <div className="movie-poster-wrapper">
                                <img src={info.poster} alt="movie-poster" className="movie-poster"/>
                            </div>
                            <div className="movie-detail">
                                <span>
                                    <span className="label">Director:</span>
                                    <span className="text">{info.director}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Script Writer:</span>
                                    <span className="text">{info.scriptwriter}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Actor:</span>
                                    <span className="text">{info.actor}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Genre:</span>
                                    <span className="text">{info.genre}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Language:</span>
                                    <span className="text">{info.language}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Initial Release Date:</span>
                                    <span className="text">{info.initialReleaseDate}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Region:</span>
                                    <span className="text">{info.region}</span>
                                </span>
                                <br/>
                                <span>
                                    <span className="label">Film Length:</span>
                                    <span className="text">{info.runtime}</span>
                                </span>
                                <br/>
                                <span>
                                    <a href={info.IMDb} >IMDb LINK</a>
                                </span>
                                <br/>
                            </div>
                            <div className="movie-rating-container">
                                <div className="rating-wrapper">
                                    <span>
                                        <img src={star} className="rating-star-icon" alt="star"></img>
                                    </span>
                                    <p className="rating-text">{info.rating}</p>/ 10
                                </div>
                                <button className="movie-add-review" onClick={addReview}>
                                    <span>
                                        <img src={pen} className="movie-add-review-icon" alt="new-review"></img>
                                    </span>
                                    <p className="movie-add-review-text">Add new review</p>
                                </button>
                            </div>
                        </div>
                        <div className="movie-brief">
                            <h2 className="movie-brief-title">Synopsis of {info.name.toUpperCase()} · · · · · ·</h2>
                            <p className="movie-brief-text">{info.brief}</p>
                        </div>
                    </div>
                </div>}
                {reviews &&
                <div className="movie-review">
                    <h2 className="movie-review-title">User Reviews</h2>
                    <ReviewList reviewData={reviews}></ReviewList>
                </div>}
            </div>
        </div>
    )
}

export default Movie;