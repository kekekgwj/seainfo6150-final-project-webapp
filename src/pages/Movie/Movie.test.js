import React from "react";
import Movie from "./Movie";

describe("Movie page tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Movie/>);
    expect(container).toMatchSnapshot();
  });
});