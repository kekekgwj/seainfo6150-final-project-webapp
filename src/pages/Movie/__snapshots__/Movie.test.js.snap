// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie page tests renders correctly 1`] = `
<div>
  <div
    class="movie-wrapper"
  >
    <div
      class="movie-content"
    />
  </div>
</div>
`;
